packages/openid4vc-client/src/OpenId4VcClientService.ts

```diff
@@ -1,10 +1,201 @@
-import { injectable, W3cCredentialService } from '@aries-framework/core'
+import type { AgentContext, Logger, W3cCredentialRecord } from '@aries-framework/core'
+import type { Jwt } from '@sphereon/openid4vci-client'
+
+import {
+  DidsApi,
+  getKeyDidMappingByVerificationMethod,
+  AgentConfig,
+  AriesFrameworkError,
+  injectable,
+  JsonEncoder,
+  JsonTransformer,
+  W3cCredentialService,
+  W3cVerifiableCredential,
+} from '@aries-framework/core'
+import {
+  Alg,
+  AuthzFlowType,
+  CredentialRequestClientBuilder,
+  OpenID4VCIClient,
+  ProofOfPossessionBuilder,
+} from '@sphereon/openid4vci-client'
+
+import { JwsService } from '../../core/src/crypto/JwsService'
+
+export interface PreAuthorizedOptions {
+  issuerUri: string
+  kid: string
+}
 
 @injectable()
 export class OpenId4VcClientService {
+  private logger: Logger
   private w3cCredentialService: W3cCredentialService
+  private jwsService: JwsService
 
-  public constructor(w3cCredentialService: W3cCredentialService) {
+  public constructor(agentConfig: AgentConfig, w3cCredentialService: W3cCredentialService, jwsService: JwsService) {
```

Comment:
You shouldn't inject the `AgentConfig` anymore as it could change based on the tenant. Maybe it's okay for the logger (but we may add context scoped loggers soon), but it probs leads to issues at some point.

We should maybe look at making the `AgentConfig` not injectable when most classes are instantiated, so this won't even be possible